tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Framework.cs

```diff
@@ -56,69 +57,138 @@ internal SecurityCoordinator(Security security, HttpContext context, Span span)
 
     private bool CanAccessHeaders => UsingIntegratedPipeline is true or null;
 
-    private static Action<string, string>? CreateThrowHttpResponseExceptionDynMeth()
+    private static Action<HttpStatusCode, string, string>? CreateThrowHttpResponseExceptionDynMeth()
     {
         try
         {
-            var exceptionType = Type.GetType("System.Web.Http.HttpResponseException, System.Web.Http");
-            if (exceptionType == null)
+            var dynMethod = new DynamicMethod(
+                "ThrowHttpResponseExceptionDynMeth",
+                typeof(void),
+                new[] { typeof(HttpStatusCode), typeof(string), typeof(string) },
+                typeof(SecurityCoordinator).Module,
+                true);
+            var il = GetBaseIlForThrowingHttpResponseException(dynMethod);
+            if (il == null)
             {
                 return null;
             }
 
-            var exceptionCtor = exceptionType.GetConstructor(new Type[] { typeof(HttpStatusCode) });
-            var exceptionResponseProperty = exceptionType.GetProperty("Response");
-
             var messageType = Type.GetType("System.Net.Http.HttpResponseMessage, System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
             if (messageType == null)
             {
                 return null;
             }
 
-            var messageContentProperty = messageType.GetProperty("Content");
-
             var contentType = Type.GetType("System.Net.Http.StringContent, System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
             if (contentType == null)
             {
                 return null;
             }
 
-            var contentCtor = contentType.GetConstructor(new Type[] { typeof(string), typeof(Encoding), typeof(string) });
-
+            // body's content
+            il.Emit(OpCodes.Ldarg_1);
+            var messageContentProperty = messageType.GetProperty("Content");
+            // StringContent(String content, Encoding, String mediaType)
+            var contentCtor = contentType.GetConstructor(new[] { typeof(string), typeof(Encoding), typeof(string) });
             var encodingType = typeof(Encoding);
             var encodingUtf8Prop = encodingType.GetProperty("UTF8");
```

Comment:
can you move this before the `Emit` blocks, to improve the readability of the IL. But leaving the comments for the arguments type.

Reply:
yes fixed in https://github.com/DataDog/dd-trace-dotnet/pull/3794/commits/109aa12c4886db146ae21e2652c8841105830022